### PR TITLE
DRIVERS-1150 Writes are not retried within transactions

### PR DESCRIPTION
**Pull request by comandeo**
Please complete the following before merging:

- [x] Update changelog.
- [ ] _Make sure there are generated JSON files from the YAML test files - not applicable._
- [ ] _Test changes in at least one language driver - not applicable._ 
- [ ] _Test these changes against all server versions and topologies (including standalone, replica set, sharded clusters, and serverless) - not applicable._

### CODE REVIEW

File: source/retryable-writes/retryable-writes.rst [hunk 1]
```diff
@@ -197,6 +197,18 @@ response.
 .. _findAndModify: https://www.mongodb.com/docs/manual/reference/command/findAndModify/
 .. _aggregate: https://www.mongodb.com/docs/manual/reference/command/aggregate/
 
+Retryable Writes Withing Transactions
```
Comment (kkloberdanz) [inline source/retryable-writes/retryable-writes.rst:200]:
```suggestion
Retryable Writes Within Transactions
```

**Review by kkloberdanz — CHANGES_REQUESTED** [review]

**Comment by kkloberdanz** [comment]
Hi Dmitry, is there any background information for this change that you would be able to share with us on this PR? Also, does it make sense to have a test for this behavior? Thank you.

File: source/retryable-writes/retryable-writes.rst [hunk 2]
```diff
@@ -838,6 +850,7 @@ inconsistent with the server and potentially confusing to developers.
 Changelog
 =========
 
+:2023-12-06: Clarify that writes are not retried with transactions.
```
Comment (isabelatkinson) [inline source/retryable-writes/retryable-writes.rst:853]:
```suggestion
:2023-12-06: Clarify that writes are not retried within transactions.
```

File: source/retryable-writes/retryable-writes.rst [hunk 3]
```diff
@@ -197,6 +197,18 @@ response.
 .. _findAndModify: https://www.mongodb.com/docs/manual/reference/command/findAndModify/
 .. _aggregate: https://www.mongodb.com/docs/manual/reference/command/aggregate/
 
+Retryable Writes Withing Transactions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In MongoDB 4.0 the only supported retryable write commands within a transaction
+are ``commitTransaction`` and ``abortTransaction``. Therefore drivers MUST NOT
+retry write commands within transactions even when ``retryWrites`` has been
+enabled on the ``MongoClient``. In addition, drivers MUST NOT add
```
Comment (isabelatkinson) [inline source/retryable-writes/retryable-writes.rst:206]:
```suggestion
set to true on the ``MongoClient``. In addition, drivers MUST NOT add
```

File: source/retryable-writes/retryable-writes.rst [hunk 4]
```diff
@@ -197,6 +197,18 @@ response.
...
+the ``RetryableWriteError`` label to any error that occurs during a write
+command within a transaction (excepting ``commitTransation``
+and ``abortTransaction``), even when ``retryWrites`` has been enabled on
```
Comment (isabelatkinson) [inline source/retryable-writes/retryable-writes.rst:209]:
```suggestion
and ``abortTransaction``), even when ``retryWrites`` has been set to true on
```

File: source/retryable-writes/retryable-writes.rst (hunk 3 above)
Comment (isabelatkinson) [reply source/retryable-writes/retryable-writes.rst:206]: this sounds a little more clear to me but feel free to accept or reject, ditto below

**Review by isabelatkinson — APPROVED** [review]
LGTM with a few suggestions, I don't need to do another round of review

@kkloberdanz I see the following comment on DRIVERS-1150:
> This is a simple but important clarification for non-engineer audiences. It requires no driver changes or new tests.

**Comment by kkloberdanz** [comment]
> > Hi Dmitry, is there any background information for this change that you would be able to share with us on this PR? Also, does it make sense to have a test for this behavior? Thank you.
> 
> Hi Kyle, as I understood [the ticket](https://jira.mongodb.org/browse/DRIVERS-1150), we just need to copy instructions from transactions spec to retryable writes spec. The behaviour is tested [here](https://github.com/mongodb/specifications/blob/master/source/transactions/tests/legacy/retryable-writes.yml#L160)

Sounds good to me, I'll approve.

**Review by kkloberdanz — APPROVED** [review]

**Pull request merged**